### ISSUE TITLE
Issues with json matcher

### ISSUE THREAD

**Issue by bboe** [sigmavirus24/betamax_matchers#1]
I'm trying to use the awesome MultipartFormDataBodyMatcher you recently wrote and am running into some issues.

On the line `request.body.encode('utf-8')`, I receive a `AttributeError: 'bytes' object has no attribute 'encode'` in python3.4 when attempting to match the existing cassette. In python 2.7 the matching exception is `UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 615: ordinal not in range(128)`. On py 2.7 the following deprecation warning is emitted (I'm not sure of its source):

```
/Users/bboe/.venv/p27/lib/python2.7/site-packages/requests-2.7.0-py2.7.egg/requests/packages/urllib3/contrib/pyopenssl.py:198: DeprecationWarning: unicode for buf is no longer accepted, use bytes
```

Below is the request that is saved in the cassette:

``` json
{
            "recorded_at": "2015-06-25T06:38:06",
            "request": {
                "body": {
                    "encoding": "utf-8",
                    "string": "--e3f99acd801b4505a9e9badc8ca979a9\r\nContent-Disposition: form-data; name=\"img_type\"\r\n\r\njpg\r\n--e3f99acd801b4505a9e9badc8ca979a9\r\nContent-Disposition: form-data; name=\"header\"\r\n\r\n1\r\n--e3f99acd801b4505a9e9badc8ca979a9\r\nContent-Disposition: form-data; name=\"r\"\r\n\r\nreddit_api_test\r\n--e3f99acd801b4505a9e9badc8ca979a9\r\nContent-Disposition: form-data; name=\"uh\"\r\n\r\nr7blffd0gne125077e056d9ac6f04a24a9ce130f8da0657878\r\n--e3f99acd801b4505a9e9badc8ca979a9\r\nContent-Disposition: form-data; name=\"api_type\"\r\n\r\njson\r\n--e3f99acd801b4505a9e9badc8ca979a9\r\nContent-Disposition: form-data; name=\"file\"; filename=\"white-square.jpg\"\r\n\r\n\ufffd\ufffd\ufffd\ufffd\u0000\u0010JFIF\u0000\u0001\u0001\u0001\u0000H\u0000H\u0000\u0000\ufffd\ufffd\u0000C\u0000\u0006\u0004\u0005\u0006\u0005\u0004\u0006\u0006\u0005\u0006\u0007\u0007\u0006\b\n\u0010\n\n\t\t\n\u0014\u000e\u000f\f\u0010\u0017\u0014\u0018\u0018\u0017\u0014\u0016\u0016\u001a\u001d%\u001f\u001a\u001b#\u001c\u0016\u0016 , #&')*)\u0019\u001f-0-(0%()(\ufffd\ufffd\u0000\u000b\b\u0001,\u0001,\u0001\u0001\u0011\u0000\ufffd\ufffd\u0000\u0015\u0000\u0001\u0001\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\b\ufffd\ufffd\u0000\u0014\u0010\u0001\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\ufffd\ufffd\u0000\b\u0001\u0001\u0000\u0000?\u0000\ufffd@\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u007f\ufffd\ufffd\r\n--e3f99acd801b4505a9e9badc8ca979a9--\r\n"
                },
                "headers": {
                    "Accept": [
                        "*/*"
                    ],
                    "Accept-Encoding": [
                        "gzip, deflate"
                    ],
                    "Connection": [
                        "keep-alive"
                    ],
                    "Content-Length": [
                        "1177"
                    ],
                    "Content-Type": [
                        "multipart/form-data; boundary=e3f99acd801b4505a9e9badc8ca979a9"
                    ],
                    "Cookie": [
                        "reddit_session=OMITTED"
                    ],
                    "User-Agent": [
                        "PRAW_test_suite PRAW/3.0.0 Python/2.7.9 Darwin-14.3.0-x86_64-i386-64bit"
                    ]
                },
                "method": "POST",
                "uri": "https://api.reddit.com/api/upload_sr_img"
            },
            "response": {
                "body": {
                    "base64_string": "H4sIAAAAAAAAA6pWSi0qyi8qVrJSiI7VUVDKzE2PLy5KVrJSUMooKSmw0tdP0ivJKM1NKtYrSk1JySzJTU3JTNRLzs/VN3Q38TVKD4jPDNANMiuK8vNNKfePzywu9fP0TIwv9a8ydPY0dIzKNogvSNbLKkhXqgUAAAD//wMAMIJgGWwAAAA=",
                    "encoding": "UTF-8"
                },
                "headers": {
                    "cache-control": [
                        "private, no-cache",
                        "no-cache"
                    ],
                    "cf-ray": [
                        "1fbeaae279190d97-SJC"
                    ],
                    "connection": [
                        "keep-alive"
                    ],
                    "content-encoding": [
                        "gzip"
                    ],
                    "content-type": [
                        "application/json; charset=UTF-8"
                    ],
                    "date": [
                        "Thu, 25 Jun 2015 06:38:05 GMT"
                    ],
                    "pragma": [
                        "no-cache"
                    ],
                    "server": [
                        "cloudflare-nginx"
                    ],
                    "transfer-encoding": [
                        "chunked"
                    ],
                    "x-content-type-options": [
                        "nosniff"
                    ],
                    "x-frame-options": [
                        "SAMEORIGIN"
                    ],
                    "x-moose": [
                        "majestic"
                    ],
                    "x-ratelimit-remaining": [
                        "292"
                    ],
                    "x-ratelimit-reset": [
                        "115"
                    ],
                    "x-ratelimit-used": [
                        "8"
                    ],
                    "x-ua-compatible": [
                        "IE=edge"
                    ],
                    "x-xss-protection": [
                        "1; mode=block"
                    ]
                },
                "status": {
                    "code": 200,
                    "message": "OK"
                },
                "url": "https://api.reddit.com/api/upload_sr_img"
            }
```

**Comment by bboe** [comment]
In this regard, if I remove the `.encode('utf-8')` part of the above line, that error goes away. Then it's simply a matching problem. The next issue is that the parts aren't sorting correctly by headers. That is resolved by prefixing `Content-Disposition` in `sorted_by_headers` as `b'Content-Disposition'`.

Once that's fixed, sadly everything matches but the upload file type itself. :( Below is a comparison of the parts that do not match:

``` python
{'content': "\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x01\x01\x00H\x00H\x00\x00\xff\xdb\x00C\x00\x06\x04\x05\x06\x05\x04\x06\x06\x05\x06\x07\x07\x06\x08\n\x10\n\n\t\t\n\x14\x0e\x0f\x0c\x10\x17\x14\x18\x18\x17\x14\x16\x16\x1a\x1d%\x1f\x1a\x1b#\x1c\x16\x16 , #&')*)\x19\x1f-0-(0%()(\xff\xc0\x00\x0b\x08\x01,\x01,\x01\x01\x11\x00\xff\xc4\x00\x15\x00\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x08\xff\xc4\x00\x14\x10\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xda\x00\x08\x01\x01\x00\x00?\x00\xaa@\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x7f\xff\xd9", 'headers': {'Content-Disposition': 'form-data; name="file"; filename="white-square.jpg"'}, 'encoding': 'utf-8'}
{'content': "\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd\x00\x10JFIF\x00\x01\x01\x01\x00H\x00H\x00\x00\xef\xbf\xbd\xef\xbf\xbd\x00C\x00\x06\x04\x05\x06\x05\x04\x06\x06\x05\x06\x07\x07\x06\x08\n\x10\n\n\t\t\n\x14\x0e\x0f\x0c\x10\x17\x14\x18\x18\x17\x14\x16\x16\x1a\x1d%\x1f\x1a\x1b#\x1c\x16\x16 , #&')*)\x19\x1f-0-(0%()(\xef\xbf\xbd\xef\xbf\xbd\x00\x0b\x08\x01,\x01,\x01\x01\x11\x00\xef\xbf\xbd\xef\xbf\xbd\x00\x15\x00\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x08\xef\xbf\xbd\xef\xbf\xbd\x00\x14\x10\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xef\xbf\xbd\xef\xbf\xbd\x00\x08\x01\x01\x00\x00?\x00\xef\xbf\xbd@\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x7f\xef\xbf\xbd\xef\xbf\xbd", 'headers': {'Content-Disposition': 'form-data; name="file"; filename="white-square.jpg"'}, 'encoding': 'utf-8'}
```

**Comment by bboe** [comment]
So perhaps the problem is how betamax is saving the contents of the file in json. It records the file as beginning with `\ufffd\ufffd\ufffd\ufffd\u0000\u0010JFIF` which I believe is UTF-8 encoded.

The file actually begins with `\xff\xd8\xff\xe0\x00\x10JFIF`. It looks like betamax is doing a unicode convert with a replace on characters it doesn't know how to handle. At that point, encoding from `utf-8` results in the following:

```
In [18]: u'\ufffd\ufffd\ufffd\ufffd\u0000\u0010JFIF`'.encode('utf-8')
Out[18]: '\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd\x00\x10JFIF`'
```

Perhaps this section also needs to be base64 encoded like the response body?

**Comment by sigmavirus24** [comment]
So yes, we probably want something like [`coerce_content`](https://github.com/sigmavirus24/betamax/blob/master/betamax/cassette/util.py#L14) in betamax but for `str.encode` instead of `str.decode`. We can also recommend that users set `preserve_exact_body_bytes=True` for these, but there's nothing the matcher can do.

Would you make a PR with the changes you've described? They all sound valid and correct, I'm just not going to have much time today. I should be able to look it over later today.

**Comment by bboe** [comment]
Yes, I will work on a PR for this issue. Thanks!

**Comment by sigmavirus24** [comment]
@bboe just noticed this hasn't been updated. Do you still have plans to tackle this or should I?

**Comment by bboe** [comment]
Thanks for checking. I completely forgot about this issue. Unfortunately, my time for non-work related open source projects has dwindled, so my need for such a PR doesn't really exist (until I find a work-related reason to need such a betamax matcher).

**Comment by bboe** [comment]
@sigmavirus24 it would seem the reason why I wrote a custom matcher to begin with was similar to this issue.

Here's code that does reproduce an issue with the jsonmatcher:

```
#!/usr/bin/env python                                                                                                                                                              
from __future__ import print_function
import json
import sys

from betamax_matchers.json_body import JSONBodyMatcher
import betamax
import requests


def main():
    print('  Python: {:>6}'.format(sys.version.split()[0]))
    print(' Betamax: {:>6}'.format(betamax.__version__))
    print('Requests: {:>6}'.format(requests.__version__))

    betamax.Betamax.register_request_matcher(JSONBodyMatcher)

    with requests.Session() as session:
        with betamax.Betamax(session, cassette_library_dir='.') as recorder:
            recorder.use_cassette(
                'test', match_requests_on=['json-body'])
            response = session.post('https://httpbin.org/post', json={'a': 2})

    import pprint
    pprint.pprint(response.json())


if __name__ == '__main__':
    sys.exit(main())
```

Output:

```
 Betamax:  0.7.2
Requests: 2.11.0
Traceback (most recent call last):
  File "./betamax_issue.py", line 28, in <module>
    sys.exit(main())
  File "./betamax_issue.py", line 21, in main
    response = session.post('https://httpbin.org/post', json={'a': 2})
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/requests/sessions.py", line 514, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/requests/sessions.py", line 471, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/requests/sessions.py", line 581, in send
    r = adapter.send(request, **kwargs)
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/betamax/adapter.py", line 123, in send
    interaction = current_cassette.find_match(request)
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/betamax/cassette/cassette.py", line 133, in find_match
    if not interaction.match(curried_matchers):
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/betamax/cassette/interaction.py", line 58, in match
    return all(m(request) for m in matchers)
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/betamax/cassette/interaction.py", line 58, in <genexpr>
    return all(m(request) for m in matchers)
  File "/Users/bboe/.venv/praw/lib/python3.5/site-packages/betamax_matchers/json_body.py", line 26, in match
    request_json = json.loads(request.body) if request.body else None
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

Since I'm stuck on this, I will look into making a PR to fix it.

**Comment by bboe** [comment]
This morning I am unable to reproduce the same issue I had with the json matcher. I'm going to close this issue for now since I don't have anything I'm working on that is currently having issue.

**Comment by bboe** [comment]
Looks like I'm getting this issue again. Running that example above in the first run I get:

```
  Python:  3.5.2
 Betamax:  0.8.0
Requests: 2.12.1
{'args': {},
 'data': '{"a": 2}',
 'files': {},
 'form': {},
 'headers': {'Accept': '*/*',
             'Accept-Encoding': 'gzip, deflate',
             'Content-Length': '8',
             'Content-Type': 'application/json',
             'Host': 'httpbin.org',
             'User-Agent': 'python-requests/2.12.1'},
 'json': {'a': 2},
 'origin': '68.6.101.75',
 'url': 'https://httpbin.org/post'}
```

A subsequent run gives me:

```
  Python:  3.5.2
 Betamax:  0.8.0
Requests: 2.12.1
Traceback (most recent call last):
  File "example.py", line 29, in <module>
    sys.exit(main())
  File "example.py", line 22, in main
    response = session.post('https://httpbin.org/post', json={'a': 2})
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/requests-2.12.1-py3.5.egg/requests/sessions.py", line 535, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/requests-2.12.1-py3.5.egg/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/requests-2.12.1-py3.5.egg/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/betamax-0.8.0-py3.5.egg/betamax/adapter.py", line 123, in send
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/betamax-0.8.0-py3.5.egg/betamax/cassette/cassette.py", line 133, in find_match
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/betamax-0.8.0-py3.5.egg/betamax/cassette/interaction.py", line 58, in match
  File "/Users/bboe/.venv/prawcore/lib/python3.5/site-packages/betamax-0.8.0-py3.5.egg/betamax/cassette/interaction.py", line 58, in <genexpr>
  File "/Users/bboe/src/betamax_matchers/betamax_matchers/json_body.py", line 31, in match
    request_json = json.loads(request.body) if request.body else None
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

I'm using the master version of betamax_matchers. Since I'm again blocked on it I will look into a fix.

**Comment by sigmavirus24** [comment]
This is _really_ weird.

**Comment by bboe** [comment]
I think the reason why I stopped getting this particular issue is I had requests fixed to 2.10 for testing:https://github.com/praw-dev/prawcore/commit/0c5d28e13c1069e00a84836d0f14daa82f624990.
